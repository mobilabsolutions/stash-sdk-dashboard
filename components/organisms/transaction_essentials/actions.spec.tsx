import Actions from './actions'
import { testRender, deepRender } from '../../../test_utils'

const actionMock = {
  isLoading: false,
  error: null,
  action: jest.fn(),
  setError: jest.fn()
}

const currency = 'EUR'
const transactionId = 'uh348274g69ch2i4tyb292-s91'

describe('Actions', () => {
  test('should show refund for authorised', () => {
    testRender(Actions, {
      refund: actionMock,
      reverse: actionMock,
      capture: actionMock,
      action: 'AUTH',
      status: 'SUCCESS',
      currency,
      transactionId,
      amount: 15.2,
      initialAmount: 1520,
      usedAmount: 0
    })
  })

  test('should show refund for partials', () => {
    testRender(Actions, {
      refund: actionMock,
      reverse: actionMock,
      capture: actionMock,
      action: 'REFUND',
      status: 'SUCCESS',
      currency,
      transactionId,
      amount: 3.2,
      initialAmount: 1520,
      usedAmount: 320
    })
  })

  test('should NOT show refund for partials if used amount is the same', () => {
    testRender(Actions, {
      refund: actionMock,
      reverse: actionMock,
      capture: actionMock,
      action: 'REFUND',
      status: 'SUCCESS',
      currency,
      transactionId,
      amount: 3.2,
      initialAmount: 1520,
      usedAmount: 1520
    })
  })

  test('should refund proper amount', async () => {
    const refund = { action: jest.fn() }
    const { getByText, getByPlaceholderText } = deepRender(Actions, {
      refund,
      reverse: actionMock,
      capture: actionMock,
      action: 'REFUND',
      status: 'SUCCESS',
      currency,
      transactionId,
      amount: 3.2,
      initialAmount: 1520,
      usedAmount: 320
    })
    getByText('Refund').click()
    expect(getByPlaceholderText('Refund').getAttribute('value')).toBe('â‚¬12.00')
  })
})
