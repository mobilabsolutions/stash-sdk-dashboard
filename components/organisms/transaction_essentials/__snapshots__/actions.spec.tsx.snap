// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should NOT show refund for partials if used amount is the same 1`] = `
<div
  className="actions__ActionContainer-zdsni1-0 ktFnMw"
/>
`;

exports[`Actions should show refund for authorised 1`] = `
<div
  className="actions__ActionContainer-zdsni1-0 ktFnMw"
>
  <button
    className="actions__ActBtn-zdsni1-1 ecmOiL flat__Button-sc-182ug78-0 jscbCh"
    label="refund"
    onClick={[Function]}
  >
    Refund
  </button>
</div>
`;

exports[`Actions should show refund for partials 1`] = `
<div
  className="actions__ActionContainer-zdsni1-0 ktFnMw"
>
  <button
    className="actions__ActBtn-zdsni1-1 ecmOiL flat__Button-sc-182ug78-0 jscbCh"
    label="refund"
    onClick={[Function]}
  >
    Refund
  </button>
</div>
`;
